### ISSUE TITLE
CellType error when ctypes is unavailable

### ISSUE THREAD

**Issue by jasonmyers** [uqfoundation/dill#150]
Possibly related to https://github.com/uqfoundation/dill/pull/1 (not sure what the current status of appengine support is)

Using dill on appengine (python 2.7), where ctypes is unavailable (so triggers the `HAS_CTYPES = False` path)

Trying to pickle an in-line function that references a variable in the outer scope causes a `cell` error.  Here is a reproducible case:

```python
# Disable ctypes -- you could also just set HAS_CTYPES = False in dill.py
import ctypes
import sys
sys.modules['ctypes'] = None

import dill

def outer():
    x = 1

    def inner():
        x

    dill.loads(dill.dumps(inner))

outer()
```

Causes the following error:

```
  File "pickletest.py", line 11, in <module>
    outer()
  File "pickletest.py", line 9, in outer
    dill.loads(dill.dumps(inner))
  File ".../dill/dill.py", line 259, in loads
    return load(file)
  File ".../dill/dill.py", line 249, in load
    obj = pik.load()
  File ".../lib/python2.7/pickle.py", line 858, in load
    dispatch[key](self)
  File ".../lib/python2.7/pickle.py", line 1083, in load_newobj
    obj = cls.__new__(cls, *args)
TypeError: object.__new__(cell) is not safe, use cell.__new__()
```

Apologies if this is intentionally unsupported at the moment.

**Comment by jasonmyers** [comment]
The above PR seems to fix the issue, but needs a sanity check

**Comment by mmckerns** [comment]
fixed by c38b68a9b6593f87cdb8365fef42512dcc1d59b5
